File: civitasJava/src/civitas/CasillaImpuesto.java
/**
 * @file CasillaImpuesto.java
 * @author Yesenia González Dávila
 * @author Esther García Gallego
 * @brief Casilla de tipo impuesto. Subclase de Casilla.
 * @note Grupo B.3
*/

package civitas;

import java.util.ArrayList;

public class CasillaImpuesto extends Casilla {
    
    /** Constructor impuesto. */
    CasillaImpuesto (String nombre, float cantidad) {
        super (nombre);
        importe = cantidad;  
        tipoCasilla = "IMPUESTO";
    }
    
    /** Recibe a un jugador en la casilla y le aplica el pago del impuesto. */
    @Override
    void recibeJugador (int actual, ArrayList<Jugador> todos) {
        if (jugadorCorrecto (actual, todos)) {
            informe (actual, todos);
            todos.get(actual).pagaImpuesto(importe);
        }
    }
}

File: civitasRuby/lib/civitas/casilla_impuesto.rb
#encoding:utf-8

=begin
Authors: Esther García Gallego
         Yesenia González Dávila
         Grupo B3
=end

module Civitas
  class CasillaImpuesto < Casilla
    
    def initialize(cantidad, nombre)
      super(nombre)
      @tituloPropiedad = nil
      @importe = cantidad
      @Carcel = -1
      @mazo = nil
      @sorpresa = nil
    end
    
    # Override 
    def recibeJugador(actual, todos)
      if (jugadorCorrecto(actual, todos))
        informe(actual, todos)
        todos[actual].pagaImpuesto(@importe)
      end
    end
    
  end
end
